pennylane/transforms/hamiltonian_expand.py

```diff
@@ -155,21 +161,32 @@ def hamiltonian_expand(tape, group=True):
 
             new_tape = new_tape.expand(stop_at=lambda obj: True)
             tapes.append(new_tape)
-    else:
-        coeffs = hamiltonian.coeffs
 
-        tapes = []
-        for o in hamiltonian.ops:
-            with tape.__class__() as new_tape:
-                for op in tape.operations:
-                    op.queue()
-                qml.expval(o)
+        def processing_fn(res_groupings):
+            dot_products = [
+                qml.math.dot(r_group, c_group)
+                for c_group, r_group in zip(coeff_groupings, res_groupings)
+            ]
+            print(coeff_groupings)
+            print(res_groupings)
+            return qml.math.sum(qml.math.stack(dot_products), axis=0)
 
-            tapes.append(new_tape)
+        return tapes, processing_fn
+
+    coeffs = hamiltonian.data
+
+    # make one tape per observable
+    tapes = []
+    for o in hamiltonian.ops:
+        with tape.__class__() as new_tape:
+            for op in tape.operations:
+                op.queue()
+            qml.expval(o)
+
+        tapes.append(new_tape)
 
+    # pylint: disable=function-redefined
     def processing_fn(res):
-        # note: res could have an extra dimension here if a shots_distribution
-        # is used for evaluation
```

Comment:
I was in two minds here, but deleted this because a developer should know everywhere that results must have an extra dimension!